### PR TITLE
Feature report requisition satisfaction rate

### PR DESCRIPTION
**Pull request by lomamech**
- Update Component Depot Select Search
- Create new component Multiple Depot Select
- Implement nez Report Satisfaction rate report of requisition
- Clean and Sanitaze

### CODE REVIEW

**Comment by lomamech** [comment]
- To use the validation rate, you must first activate the authorization in the configuration of actions related to user roles
image

The Report interface is very simple and invites the user to select one or more depots as well as the search period
image

image

image

The report first lists the depots that have been selected and the time range

The report presents the requests for requisitions made during the period by displaying the reference number, the reference of the last stock movement, the quantity requisitioned, the quantity validated and the quantity delivered, the delivery rate

And for the beneficiary and supplier depots, the system displays the overall satisfaction rate by quantity but also by number of items

**Review by jmcameron — COMMENTED** [review]
I noticed that 
- `ENABLE_REQUISITION_VALIDATION_STEP`
- `ENABLE_REQUISITION_VALIDATION_STEP_HELP`

appear twice in client/src/i18n/en/stock.json.   I think you want to keep the ones at lines 333 and 334 and remove the ones on lines 372 and 373.

I tested with bhima_test, which seemed to work but I did not see the correct calculations in the table(s) in the report because I was not quite sure how to do the validation step.  I presume the calculations are correct based on the snapshot in the PR.   I suggest setting the background for the regular cells of the tables to white (transparent) and keeping gray for titles and row/column headers.  I did not have time to download and test with the SEMI data.

The code looks okay.

This PR looks okay to me, although you should be aware that my testing was minimal.

**Comment by lomamech** [comment]
@jmcameron @mbayopanda this PR is ready for the new review

**Comment by lomamech** [comment]
@mbayopanda  I just noticed that you had to work a lot on the budget
image

**Review by jmcameron — COMMENTED** [review]
I just tested this PR again and it seems to work.  I could not get the final "delivered" counts to work with BHIMA.  I am pretty sure that is because I am testing it with bhima_test data and I am missing a step somewhere.  (I did dispense some of the stock shown to patients.)  I'm confident that the screenshots that Chris shows in the PR indicate that it works.

image

As far as I can tell, I think this PR is ready to merge.

**Comment by lomamech** [comment]
> I just tested this PR again and it seems to work. I could not get the final "delivered" counts to work with BHIMA. I am pretty sure that is because I am testing it with bhima_test data and I am missing a step somewhere. (I did dispense some of the stock shown to patients.) I'm confident that the screenshots that Chris shows in the PR indicate that it works.
> 
> image
> 
> As far as I can tell, I think this PR is ready to merge.



> I just tested this PR again and it seems to work. I could not get the final "delivered" counts to work with BHIMA. I am pretty sure that is because I am testing it with bhima_test data and I am missing a step somewhere. (I did dispense some of the stock shown to patients.) I'm confident that the screenshots that Chris shows in the PR indicate that it works.
> 
> image
> 
> As far as I can tell, I think this PR is ready to merge.

There remains the stock issue step from a requisition, you just have to select the supplier depot, then the beneficiary depot and select the requisition

File: client/src/i18n/en/report.json [hunk 1]
```diff
@@ -220,6 +220,7 @@
       "TOTAL_DAYS" : "Total Days",
       "RATIO" : "Ratio"
     },
+    "SATISFACTION_RATE_REPORT" : "Satisfaction rate report",
```
Comment (jmcameron) [inline client/src/i18n/en/report.json:223]: This should be capitalized, like the other report names (in English).

**Review by jmcameron — COMMENTED** [review]
I suggested a minor spelling change.

I tested this again with bhima_test and got the same results:  I can never get anything but 0 in the "Quantity Delivered" column.  

As Bruce suggested, I encourage Chris to write step-by-step directions to get this to work with bhima_test.   I am missing a step and I tried following the directions in the PR and doing different actions several times but could not get the "Quantity Delivered" to work.

**Review by jmcameron — COMMENTED** [review]
I tested this again and it worked.  I had not done the validation step with the requisition.  So I think it is ready to merge.

**Pull request merged**